### PR TITLE
- Refactoring CollectWgsMetrics in order to be able to easily create

### PR DESCRIPTION
**Pull request by yfarjoun**
CollectRawWGSMetrics.
- Create RawWGSMetrics by overriding some default option values and
  repackaging the metrics class.
- Moved CollectIlluminaSummaryMetrics (will be deleted eventually)
- Fixed some test files. (unused)
- Removed the buggy CLP CollectIlluminaSummaryMetrics.java and its test

### CODE REVIEW

File: src/java/picard/analysis/CollectWgsMetrics.java [hunk 1]
```diff
@@ -170,14 +174,14 @@ protected int doWork() {
...
-                    continue;
-                }
-                if (!readNames.add(recs.getRecord().getReadName())) {
-                    ++basesExcludedByOverlap;
-                    continue;
+
+                if (recs.getBaseQuality() < MINIMUM_BASE_QUALITY)                   { ++basesExcludedByBaseq;   continue; }
+                if (!readNames.add(recs.getRecord().getReadName()))                 { ++basesExcludedByOverlap; continue; }
+                pileupSize++;
+                if (pileupSize <= max) {
```
Comment (nh13) [inline src/java/picard/analysis/CollectWgsMetrics.java]: I am unclear why we need this if statement here?

File: src/java/picard/analysis/CollectWgsMetrics.java [hunk 2]
```diff
@@ -185,6 +189,7 @@ protected int doWork() {
             if (depth < readNames.size()) basesExcludedByCapping += readNames.size() - max;
             HistogramArray[depth]++;
 
+
```
Comment (nh13) [inline src/java/picard/analysis/CollectWgsMetrics.java]: ! ! !

File: src/java/picard/analysis/CollectWgsMetrics.java [hunk 3]
```diff
@@ -233,10 +244,17 @@ protected int doWork() {
         final MetricsFile<WgsMetrics, Integer> out = getMetricsFile();
         out.addMetric(metrics);
         out.addHistogram(histo);
+        if(INCLUDE_BQ_HISTOGRAM) {
```
Comment (nh13) [inline src/java/picard/analysis/CollectWgsMetrics.java]: ```if (```

**Comment by nh13** [comment]
:+1:

File: src/java/picard/analysis/CollectWgsMetrics.java (hunk 1 above)
Comment (yfarjoun) [inline src/java/picard/analysis/CollectWgsMetrics.java]: see line 153/156...index out of bounds...

Comment (yfarjoun) [inline src/java/picard/analysis/CollectWgsMetrics.java]: If pileupsize>max, the base will not be counted (in the next step), and therefore I do not want to add its quality to the BQ distribution.

Comment (nh13) [inline src/java/picard/analysis/CollectWgsMetrics.java]: Lets discuss in person, I don't understand.

**Comment by yfarjoun** [comment]
responded to reviewers comments and rebased on top of jay's commit (which this needs)

**Pull request merged**